src/elements/content-explorer/Content.js

```diff
@@ -46,65 +47,79 @@ type Props = {
     rootId: string,
     tableRef: Function,
     view: View,
+    viewMode?: ViewMode,
 };
 
 const Content = ({
-    view,
-    isSmall,
-    isMedium,
-    isTouch,
-    rootId,
-    rootElement,
-    currentCollection,
-    tableRef,
-    focusedRow,
-    canDownload,
     canDelete,
+    canDownload,
+    canPreview,
     canRename,
     canShare,
-    canPreview,
+    currentCollection,
+    focusedRow,
+    isMedium,
+    isSmall,
+    isTouch,
     onItemClick,
-    onItemSelect,
     onItemDelete,
     onItemDownload,
+    onItemPreview,
     onItemRename,
+    onItemSelect,
     onItemShare,
-    onItemPreview,
     onSortChange,
-}: Props) => (
-    <div className="bce-content">
-        {view === VIEW_ERROR || view === VIEW_SELECTED ? null : (
-            <ProgressBar percent={currentCollection.percentLoaded} />
-        )}
-        {isEmpty(view, currentCollection) ? (
-            <EmptyState view={view} isLoading={currentCollection.percentLoaded !== 100} />
-        ) : (
-            <ItemList
-                view={view}
-                isSmall={isSmall}
-                isMedium={isMedium}
-                isTouch={isTouch}
-                rootId={rootId}
-                rootElement={rootElement}
-                focusedRow={focusedRow}
-                currentCollection={currentCollection}
-                tableRef={tableRef}
-                canShare={canShare}
-                canPreview={canPreview}
-                canDelete={canDelete}
-                canRename={canRename}
-                canDownload={canDownload}
-                onItemClick={onItemClick}
-                onItemSelect={onItemSelect}
-                onItemDelete={onItemDelete}
-                onItemDownload={onItemDownload}
-                onItemRename={onItemRename}
-                onItemShare={onItemShare}
-                onItemPreview={onItemPreview}
-                onSortChange={onSortChange}
-            />
-        )}
-    </div>
-);
+    rootElement,
+    rootId,
+    tableRef,
+    view,
+    viewMode = VIEW_MODE_LIST,
+}: Props) => {
+    let content;
+    switch (viewMode) {
+        case VIEW_MODE_GRID:
+            content = <div className="bdl-GridView">Placeholder for grid view</div>;
```

Comment:
It seems like the props passed to ItemList and GridView may end up being very similar. Would something like this work?

```
const View = viewMode === VIEW_MODE_LIST ? ItemList : GridView;
...
<View 
    isSmall={isSmall}
    isMedium={isMedium}
    ...etc
/>
```

Reply:
That would work, but GridView (and perhaps some additional view) may need to be wrapped in an `AutoSizer` or some other component.  I think to build for the future and not let things get too hairy, I'll opt for doing a `ContentView` component.